Realm/RLMResults.mm

```diff
@@ -364,6 +364,32 @@ - (RLMResults *)sortedResultsUsingDescriptors:(NSArray<RLMSortDescriptor *> *)pr
     });
 }
 
+- (RLMResults *)distinctResultsUsingKeyPaths:(NSArray<NSString *> *)keyPaths {
+    if (keyPaths.count == 0) {
+        @throw RLMException(@"Must specify at least one keypath.");
+    }
+    for (NSString *keyPath in keyPaths) {
+        if ([keyPath containsString:@"@"]) {
+            @throw RLMException(@"Cannot distinct on keypath '%@': KVC collection operators are not supported.", keyPath);
+        }
+    }
+    
+    return translateRLMResultsErrors([&] {
+        if (_results.get_mode() == Results::Mode::Empty) {
+            return self;
+        }
+        
+        __block std::vector<std::string> kp;
```

Comment:
`__block` is unnecessary here.

I don't like using `kp` for a name here, but I don't have any other great ideas for how to otherwise distinguish this from `keyPaths` 🤷‍♂️

Reply:
`keyPathVector`? Verbose, but clear.

Reply:
Yeah, I'd be fine with that.

Reply:
Went with `keyPathsVector`...plural `keyPaths` may be redundant given that it is a vector...